common/archiver/provider/provider.go

```diff
@@ -53,6 +54,8 @@ type (
 	}
 
 	archiverProvider struct {
+		m *sync.RWMutex
```

Comment:
You can just make archiverProvider sync.RWMutex then you can do p.RLock and p.RUnlock

Reply:
Updated.